test/src/611-geoservice/main

```diff
@@ -0,0 +1,97 @@
+cvmfs_test_name="GeoIP Service"
+cvmfs_test_autofs_on_startup=false
+
+cvmfs_getaddr() {
+  # exit after the first line in case there are multiple addresses
+  ip $1 -o addr show scope global|awk '{split($4,a,"/");print a[1];exit}'
+}
+
+cvmfs_check_georesult() {
+  local result=$1
+  local me=$2
+  local reqorder=$3
+  local expectorder
+  echo "  result for $reqorder is $result"
+  [ -n "$1" ] || return 1
+  if [[ "$me" == *.cern.ch ]] || [[ "$me" == *.fnal.gov ]]; then
+    local cern
+    local fnal
+    local asgc
+    local n=1
+    for site in ${reqorder//,/ }; do
+      eval $site=$n
+      let n+=1
+    done
+    if [[ "$me" == *.cern.ch ]]; then
+      expectorder=$cern,$fnal,$asgc
+    elif [[ "$me" == *.fnal.gov ]]; then
+      expectorder=$fnal,$cern,$asgc
+    fi
+  else
+    # not testing at CERN or FNAL, so any order will do
+    expectorder="[123],[123],[123]"
+  fi
+  [[ $result == $expectorder ]] || return 2
+}
+
+
+cvmfs_run_test() {
+  logfile=$1
+
+  echo "checking if both IPv4 and IPv6 addresses are available"
+  local ipv4addr="`cvmfs_getaddr -4`"
+  local ipv6addr="`cvmfs_getaddr -6`"
+  [ -n "$ipv4addr" ] && [ -n "$ipv6addr" ] || return 1
```

Comment:
The test fails on non dual stack machines.  I think this is a bit strict, the test should rather quit with a warning.